src/PAModel/Serializers/MsAppSerializer.cs

```diff
@@ -527,6 +549,25 @@ private static IEnumerable<FileEntry> GetMsAppFiles(this CanvasDocument app, Err
             var props = app._properties.JsonClone();
             if (app._connections != null)
             {
+                var connectionIDReferences = app._entropy.LocalConnectionIDReferences;
+                if (connectionIDReferences != null)
+                {
+                    foreach (var connection in app._connections)
+                    {
+                        if (connectionIDReferences.TryGetValue(connection.Key, out string instanceID))
+                        {
+                            var extensionData = connection.Value.ExtensionData;
+                            JsonElement connectionInstanceID = JsonSerializer.Deserialize<JsonElement>(instanceID);
+
+                            // Deserialized conn instance id is added to the extension data to eventually add it back to properties.json while packing
+                            extensionData.Add(ConnectionInstanceIDPropertyName, connectionInstanceID);
+
+                            // Making sure instance references are removed after adding it to the extension data
+                            connectionIDReferences.Remove(connection.Key);
+                        }
+                    }
+                    app._entropy.LocalConnectionIDReferences = null;
```

Comment:
Just curious, why do we need to `null` this out?

Reply:
Looks like this line is not needed since key values are removed from connections(line:566) after adding it back to properties. Went ahead and removed it. Thanks for catching it.